lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1670,14 +1669,17 @@ synthesizeCoroutineAccessorBody(AccessorDecl *accessor, ASTContext &ctx) {
 
   // If this is a variable with an attached property wrapper, then
   // the accessors need to yield the wrappedValue or projectedValue.
-  if (auto var = dyn_cast<VarDecl>(storage)) {
-    if (var->hasAttachedPropertyWrapper()) {
-      target = TargetImpl::Wrapper;
-    }
+  if (storage->getReadImpl() == ReadImplKind::Read ||
+      storageReadWriteImpl == ReadWriteImplKind::Modify) {
```

Comment:
I'm not sure this is the right condition. Don't you want something like "accessor->getAccessorKind() == AccessorKind::Read || storageReadWriteImpl == ReadWriteImplKind::Modify)"?